### PR TITLE
Create one more goal" checkbox in GoalCreateForm

### PR DESCRIPTION
**Pull request by Troff8**
## PR includes

- [x] Feature

## Related issues

Resolve [#717](https://github.com/taskany-inc/issues/issues/717)

## Screenshots
image

### CODE REVIEW

File: src/components/GoalCreateForm/GoalCreateForm.i18n/ru.json [hunk 1]
```diff
@@ -2,5 +2,9 @@
     "New goal": "Новая цель",
     "Pro tip!": "Держи совет!",
     "Press key to create the goal": "Нажми {key}, чтобы создать цель.",
-    "Create goal": "Создать цель"
+    "Create & Go": "Создать и перейти",
+    "Create one more": "Создать еще",
+    "Create": "Создать",
+    "Create and go to the goal page": "Создать и перейти на страницу цели",
+    "Create and open new form for the next goal": "Создайте и откройте новую форму для следующей цели"
```
Comment (awinogradov) [inline src/components/GoalCreateForm/GoalCreateForm.i18n/ru.json:9]: `Создать и открыть новую форму`

File: src/components/GoalCreateForm/GoalCreateForm.tsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ import { Tip } from '../Tip';
 import { Keyboard } from '../Keyboard';
 import { GoalForm } from '../GoalForm/GoalForm';
 import { trpc } from '../../utils/trpcClient';
-import { GoalCommon, goalCommonSchema } from '../../schema/goal';
+import { GoalCommon, goalCommonSchema, dependencyKind } from '../../schema/goal';
```
Comment (awinogradov) [inline src/components/GoalCreateForm/GoalCreateForm.tsx:14]: Useless import of `dependencyKind`

File: src/components/GoalCreateForm/GoalCreateForm.tsx [hunk 3]
```diff
@@ -32,7 +32,38 @@ const GoalCreateForm: React.FC = () => {
     const [currentProjectCache] = useLocalStorage('currentProjectCache');
     const [recentProjectsCache, setRecentProjectsCache] = useLocalStorage('recentProjectsCache', {});
     const [busy, setBusy] = useState(false);
+    const [actionBtnText, setActionBtnText] = useState<'Create & Go' | 'Create one more' | 'Create'>('Create & Go');
```
Comment (awinogradov) [inline src/components/GoalCreateForm/GoalCreateForm.tsx:35]: Type `string`.

File: src/components/GoalCreateForm/GoalCreateForm.tsx [hunk 4]
```diff
@@ -32,7 +32,38 @@ const GoalCreateForm: React.FC = () => {
...
+            clue: tr('Create and open new form for the next goal'),
+            value: 2,
+        },
+        {
+            title: tr('Create'),
+            value: 3,
+        },
+    ];
+
+    type textButtonsItems = {
```
Comment (awinogradov) [inline src/components/GoalCreateForm/GoalCreateForm.tsx:55]:
``` ts
type buttonItems = typeof textButtons;

(item: buttonItems[number]) => {...}
```

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/components/GoalCreateForm/GoalCreateForm.tsx [hunk 5]
```diff
@@ -49,12 +80,24 @@ const GoalCreateForm: React.FC = () => {
                     cache: form.parent,
                 };
             }
-
             setRecentProjectsCache(newRecentProjectsCache);
             setLastProjectCache(form.parent);
 
-            router.goal(res._shortId);
-            dispatchModalEvent(ModalEvent.GoalCreateModal)();
+            if (id === 1) {
```
[the remark this answers is not in the document]
Comment (Troff8) [reply src/components/GoalCreateForm/GoalCreateForm.tsx:86]: переименовать id в goalType ?

[the remark this answers is not in the document]
Comment (awinogradov) [reply src/components/GoalCreateForm/GoalCreateForm.tsx:86]: createGoalType

File: src/components/GoalCreateForm/GoalCreateForm.tsx (hunk 3 above)
Comment (Troff8) [reply src/components/GoalCreateForm/GoalCreateForm.tsx:35]: тогда будет - Argument of type 'string' is not assignable to parameter of type '"Create & Go" | "New goal" | "Pro tip!" | "Press key to create the goal" | "Create one more" | "Create" | "Create and go to the goal page" | "Create and open new form for the next goal"'.
вот здесь - actionBtnText={tr(actionBtnText)}

File: src/components/GoalCreateForm/GoalCreateForm.tsx [hunk 6]
```diff
@@ -25,14 +36,48 @@ const StyledFormBottom = styled.div`
...
+            clue: tr('Create and open new form for the next goal'),
+            value: 2,
+        },
+        {
+            title: tr('Create'),
+            value: 3,
+        },
+    ];
+
+    type buttonItems = typeof createOptions;
```
Comment (awinogradov) [inline src/components/GoalCreateForm/GoalCreateForm.tsx:76]:
``` ts
type CreateOptions
```

File: src/components/GoalForm/GoalForm.tsx [hunk 7]
```diff
@@ -258,15 +247,7 @@ export const GoalForm: React.FC<GoalFormProps> = ({
...
-                        <FormAction right inline>
-                            <Button
-                                view="primary"
-                                disabled={busy}
-                                outline={!isValid}
-                                type="submit"
-                                text={actionBtnText}
-                            />
-                        </FormAction>
+                        {renderActionButton({ busy: busy ?? false, isValid: isValid ?? false })}
```
Comment (awinogradov) [inline src/components/GoalForm/GoalForm.tsx:250]:
``` ts
renderActionButton({ busy: Boolean(busy), isValid: Boolean(isValid)
```

**Review by awinogradov — CHANGES_REQUESTED** [review]

**Pull request merged**